linkerd/announcer/curatorsd/src/main/scala/com/medallia/l5d/curatorsd/announcer/CuratorSDAnnouncer.scala

```diff
@@ -0,0 +1,81 @@
+package com.medallia.l5d.curatorsd.announcer
+
+import java.net.InetSocketAddress
+
+import com.medallia.l5d.curatorsd.common.CuratorSDCommon
+import com.medallia.servicediscovery.ServiceInstanceInfo
+import com.twitter.finagle.{Announcement, Path}
+import com.twitter.logging.Logger
+import com.twitter.util.{Future, Time}
+import io.buoyant.linkerd.FutureAnnouncer
+import org.apache.curator.x.discovery._
+
+/**
+ * Announcer that uses the curator service discovery format.
+ * <p>
+ * Format used for the service name Path: /#/com.medallia.curatorsd/{full service name}[#{tenant}]
+ * tenant is optional. If it's empty or the last part is missing, it will be registered as a multi-tenant service.
+ */
+class CuratorSDAnnouncer(zkConnectStr: String) extends FutureAnnouncer {
+
+  override val scheme: String = "zk-curator"
+
+  private val log = Logger(getClass)
+
+  val serviceDiscoveryInfo = CuratorSDCommon.createServiceDiscovery(zkConnectStr)
+
+  private def announce(serviceId: String, tenant: Option[String], address: InetSocketAddress): Future[Announcement] = {
+    val tenantStr = tenant.getOrElse("(multi-tenant)")
+    log.info("Announcing %s, tenant: %s address: %s, ZK cluster: %s", serviceId, tenantStr, address, zkConnectStr)
+
+    val serviceFullPath = CuratorSDCommon.getServiceFullPath(serviceId, tenant)
+
+    // TODO (future) how to specify https? Handle this when we work on the Namer.
+    val builder = ServiceInstance.builder[ServiceInstanceInfo]
+      .name(serviceFullPath)
+      .uriSpec(new UriSpec(s"http://$address.getHostString:$address.getPort"))
+      .port(address.getPort)
+      .address(address.getHostString)
+      .payload(ServiceInstanceInfo(s"serviceId: $serviceId, tenant: $tenantStr"))
+      .serviceType(ServiceType.DYNAMIC)
+
+    val serviceInstance = builder.build
+
+    serviceDiscoveryInfo.serviceDiscovery.registerService(serviceInstance)
```

Comment:
Does service discovery throw an exception when zookeeper is not available or mis-configured?